libtenzir/src/tql2/eval_unary.cpp

```diff
@@ -88,6 +88,34 @@ struct EvalUnOp<ast::unary_op::neg, T> {
   }
 };
 
+template <>
+struct EvalUnOp<ast::unary_op::neg, duration_type> {
+  using T = type_to_arrow_array_t<duration_type>;
```

Comment:
I would remove this alias and just use `arrow::DurationArray` directly instead of `T`.